app/code/Magento/Downloadable/Test/Mftf/Test/SelectAllDownloadableLinksDownloadableProductTest.xml

```diff
@@ -0,0 +1,101 @@
+                                        parameterArray="[$$createCategory.name$$]" stepKey="fillCategory"/>
+
+            <!-- Fill downloadable link information before the creation link -->
+            <actionGroup ref="AdminAddDownloadableLinkInformationActionGroup" stepKey="addDownloadableLinkInformation"/>
+
+            <!-- Links can be purchased separately -->
+            <checkOption selector="{{AdminProductDownloadableSection.isLinksPurchasedSeparately}}"
+                         stepKey="checkOptionPurchaseSeparately"/>
+
+            <!-- Add first downloadable link -->
+            <actionGroup ref="addDownloadableProductLinkWithMaxDownloads" stepKey="addFirstDownloadableProductLink">
+                <argument name="link" value="downloadableLinkWithMaxDownloads"/>
+            </actionGroup>
+
+            <!-- Add second downloadable link -->
+            <actionGroup ref="addDownloadableProductLink" stepKey="addSecondDownloadableProductLink">
+                <argument name="link" value="downloadableLink"/>
+            </actionGroup>
+
+            <!-- Save product -->
+            <actionGroup ref="saveProductForm" stepKey="saveProduct"/>
+        </before>
+        <after>
+            <!-- Delete category -->
+            <deleteData createDataKey="createCategory" stepKey="deleteCategory"/>
+
+            <!-- Delete created downloadable product -->
+            <actionGroup ref="deleteProductUsingProductGrid" stepKey="deleteProduct">
+                <argument name="product" value="DownloadableProduct"/>
+            </actionGroup>
+
+            <!-- Log out -->
+            <actionGroup ref="logout" stepKey="logout"/>
+        </after>
+
+        <!-- Step 1: Navigate to store front Product page as guest -->
+        <amOnPage url="/{{DownloadableProduct.sku}}.html"
+                  stepKey="amOnStorefrontProductPage"/>
+
+        <!-- Step 2: click on select all checkbox -->
+        <click
+            selector="{{StorefrontDownloadableProductSection.downloadableLinkSelectAllCheckbox"
+            stepKey="selectAllProductLink"/>
+
+        <!-- Step 3: Make sure that all product links are checked -->
+        <seeCheckboxIsChecked selector="{{StorefrontDownloadableProductSection.downloadableLinkByTitle(downloadableLinkWithMaxDownloads.title)}}" stepKey="seeFirstCheckboxChecked"/>
+
+        <seeCheckboxIsChecked selector="{{StorefrontDownloadableProductSection.downloadableLinkByTitle(downloadableLink.title)}}" stepKey="seeSecondCheckboxChecked"/>
+
+        <!-- Step 4: click again on select all checkbox -->
+        <click
+            selector="{{StorefrontDownloadableProductSection.downloadableLinkSelectAllCheckbox"
+            stepKey="unselectAllProductLink"/>
+
+        <!-- Step 5: Make sure that all product links are unchecked -->
+        <uncheckOption selector="{{StorefrontDownloadableProductSection.downloadableLinkByTitle(downloadableLinkWithMaxDownloads.title)}}" stepKey="seeFirstCheckboxUnChecked"/>
```

Comment:
Looks like there should be `dontSeeCheckboxIsChecked`

Reply:
Now you updated `stepKey`, I don't mind about it. But I thought that `<uncheckOption ...>`  should be replaced to `<dontSeeCheckboxIsChecked ...>`  because right now test will do the action "uncheck option", while we want to check that it's actually not checked